app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,76 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, except: %i[index show]
```

Comment:
- [ ] it uses `before_action` to restrict access to the `:new` and `:create` methods (uses ApplicationController helper method) ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png) from [Odin](https://www.theodinproject.com/courses/ruby-on-rails/lessons/authentication))